ludwig/api.py

```diff
@@ -332,6 +334,23 @@ def __init__(
         # online training state
         self._online_trainer = None
 
+        # Zero-shot LLM usage.
+        if (
+            self.config_obj.model_type == MODEL_LLM
+            and self.config_obj.trainer.type == "none"
+            # Category output features require a vocabulary. The LLM LudwigModel should be initialized with
+            # model.train(dataset).
+            and self.config_obj.output_features[0].type == "text"
+        ):
+            self._initialize_llm()
+
+    def _initialize_llm(self, random_seed: int = default_random_seed):
+        """Initialize the LLM model.
+
+        Should only be used in a zero-shot (NoneTrainer) setting.
+        """
+        self.model = LudwigModel.create_model(self.config_obj, random_seed=random_seed)
```

Comment:
It's possible that at the end of `LudwigModel.create()` in `_initialize_llm`, the actual `self.model.model` is on CPU.  Specifically, this is the case when quantization isn't being used to load the model. (https://github.com/ludwig-ai/ludwig/blob/master/ludwig/models/llm.py#L83C1-L87C43), specifically, the `device_map` keyword decides whether it is on CPU or GPU. If not set to `auto` (in our case, when there's no quantization), it ends up being loaded on CPU instead of GPU. 

It's not wrong necessarily to leave the model on CPU and call generate, but I'd double check that `self.model.model.device` is 'cuda', and if not, I'd move it to CUDA because generation will be much faster that way.

Reply:
Ooh, good call out and good to know.

I added a check to move the model to cuda if the model device is CPU and tested this out with a couple of different non-quantized LLMs.

```
Loading large language model...
Done.
Loaded HuggingFace implementation of facebook/opt-350m tokenizer
LLM was initialized on cpu. Moving to GPU for inference.
Finished generating in: 0.57s.
Finished generating in: 0.16s.
Finished generating in: 0.31s.
Finished generating in: 0.61s.
Finished generating in: 1.24s.
Finished generating in: 2.47s.
Finished generating in: 5.04s.
```